pkg/mtbroker/filter/filter_handler.go

```diff
@@ -0,0 +1,388 @@
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	"go.uber.org/zap"
+	eventingv1alpha1 "knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+	"knative.dev/eventing/pkg/broker"
+	eventinglisters "knative.dev/eventing/pkg/client/listers/eventing/v1alpha1"
+	"knative.dev/eventing/pkg/kncloudevents"
+	"knative.dev/eventing/pkg/logging"
+	"knative.dev/eventing/pkg/reconciler/trigger/path"
+	"knative.dev/eventing/pkg/utils"
+	pkgtracing "knative.dev/pkg/tracing"
+)
+
+const (
+	writeTimeout = 15 * time.Minute
+
+	passFilter FilterResult = "pass"
+	failFilter FilterResult = "fail"
+	noFilter   FilterResult = "no_filter"
+
+	// readyz is the HTTP path that will be used for readiness checks.
+	readyz = "/readyz"
+
+	// TODO make these constants configurable (either as env variables, config map, or part of broker spec).
+	//  Issue: https://github.com/knative/eventing/issues/1777
+	// Constants for the underlying HTTP Client transport. These would enable better connection reuse.
+	// Set them on a 10:1 ratio, but this would actually depend on the Triggers' subscribers and the workload itself.
+	// These are magic numbers, partly set based on empirical evidence running performance workloads, and partly
+	// based on what serving is doing. See https://github.com/knative/serving/blob/master/pkg/network/transports.go.
+	defaultMaxIdleConnections        = 1000
+	defaultMaxIdleConnectionsPerHost = 100
+)
+
+// Handler parses Cloud Events, determines if they pass a filter, and sends them to a subscriber.
+type Handler struct {
+	logger        *zap.Logger
+	triggerLister eventinglisters.TriggerLister
+	ceClient      cloudevents.Client
+	reporter      StatsReporter
+	isReady       *atomic.Value
+}
+
+type sendError struct {
+	Err    error
+	Status int
+}
+
+func (e sendError) Error() string {
+	return e.Err.Error()
+}
+
+func (e sendError) Unwrap() error {
+	return e.Err
+}
+
+// FilterResult has the result of the filtering operation.
+type FilterResult string
+
+// NewHandler creates a new Handler and its associated MessageReceiver. The caller is responsible for
+// Start()ing the returned Handler.
+func NewHandler(logger *zap.Logger, triggerLister eventinglisters.TriggerLister, reporter StatsReporter) (*Handler, error) {
+	httpTransport, err := cloudevents.NewHTTPTransport(cloudevents.WithBinaryEncoding(), cloudevents.WithMiddleware(pkgtracing.HTTPSpanIgnoringPaths(readyz)))
+	if err != nil {
+		return nil, err
+	}
+
+	connectionArgs := kncloudevents.ConnectionArgs{
+		MaxIdleConns:        defaultMaxIdleConnections,
+		MaxIdleConnsPerHost: defaultMaxIdleConnectionsPerHost,
+	}
+	ceClient, err := kncloudevents.NewDefaultClientGivenHttpTransport(httpTransport, &connectionArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	r := &Handler{
+		logger:        logger,
+		triggerLister: triggerLister,
+		ceClient:      ceClient,
+		reporter:      reporter,
+		isReady:       &atomic.Value{},
+	}
+	r.isReady.Store(false)
+
+	httpTransport.Handler = http.NewServeMux()
```

Comment:
We should expose port here as well.

In my experimenting I used `FILTER_PORT` for this, and `INGRESS_PORT` for the other.